htlcswitch/link.go

```diff
@@ -1247,8 +1248,117 @@ func (l *channelLink) String() string {
 // another peer or if the update was created by user
 //
 // NOTE: Part of the ChannelLink interface.
-func (l *channelLink) HandleSwitchPacket(packet *htlcPacket) {
-	l.mailBox.AddPacket(packet)
+func (l *channelLink) HandleSwitchPacket(pkt *htlcPacket) error {
```

Comment:
I know we discussed this earlier, but what's the benefit of moving the circuit handling logic _into_ the link, rather than keeping it at the switch? Logically, seems that readability of control flow would be kept in tack by handling all the circuit logic, _then_ calling `HandleSwitchPacket`. Also FWIW, we also need to do an extra set of type switches with this change.

Reply:
This has been returned to its original form that just adds the packet to the link's mailbox. The calls to add the circuits now happens in `forward/forwardBatch` and calls to teardown circuits happen in `handlePacketForward` and `handleLocalDispatch`